src/editor/tools/scenario_worker_tool.cc

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2002-2019 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+
+#include "editor/tools/scenario_worker_tool.h"
+
+#include "editor/editorinteractive.h"
+#include "logic/map_objects/tribes/ship.h"
+#include "logic/map_objects/tribes/worker.h"
+#include "logic/player.h"
+#include "logic/widelands_geometry.h"
+
+int32_t ScenarioPlaceWorkerTool::handle_click_impl(const Widelands::NodeAndTriangle<>& center,
+                                                   EditorInteractive& eia,
+                                                   EditorActionArgs* args,
+                                                   Widelands::Map* map) {
+	if (args->new_owner < 1 || args->new_owner > map->get_nrplayers() ||
+	    args->worker_types.empty()) {
+		return 0;
+	}
+	const size_t nr_items = args->worker_types.size();
+	Widelands::EditorGameBase& egbase = eia.egbase();
+	args->infrastructure_placed = 0;
+	if (args->random_index < 0) {
```

Comment:
Use a `constexpr` for this. You can use numeric limits max instead of `-1` and then have an unsigned data type.